backend/src/main/java/ai/verta/modeldb/utils/RdbmsUtils.java

```diff
@@ -1720,23 +1722,38 @@ public static Order getOrderBasedOnSortKey(
                 }
               }
               predicate = predicate.toBuilder().setOperator(operator).build();
-              if (key.equalsIgnoreCase("owner")
-                  && (operator.equals(Operator.CONTAIN) || operator.equals(Operator.NOT_CONTAIN))) {
-                Predicate fuzzySearchPredicate =
-                    getFuzzyUsersQueryPredicate(
-                        authService, roleService, builder, entityRootPath, predicate);
-                if (fuzzySearchPredicate != null) {
-                  keyValuePredicates.add(fuzzySearchPredicate);
+              if (key.equalsIgnoreCase(ModelDBConstants.OWNER)) {
+                if ((operator.equals(Operator.CONTAIN) || operator.equals(Operator.NOT_CONTAIN))) {
+                  Predicate fuzzySearchPredicate =
+                      getFuzzyUsersQueryPredicate(
+                          authService,
+                          roleService,
+                          builder,
+                          entityRootPath,
+                          predicate,
+                          modelDBServiceResourceTypes);
+                  if (fuzzySearchPredicate != null) {
+                    keyValuePredicates.add(fuzzySearchPredicate);
+                  } else {
+                    throw new ModelDBException(
+                        ModelDBConstants.INTERNAL_MSG_USERS_NOT_FOUND, Code.FAILED_PRECONDITION);
```

Comment:
Bad request?

Reply:
this is for internal purpose. we are throwing from here and it will return empty response after catch it from service https://github.com/VertaAI/modeldb/blob/40c3035419b03bbda87cea130d4b9fab95f51923/backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java#L1435

Reply:
Oh man, this looks very brittle. We can fix it later.

Reply:
Ok.